server/config.go

```diff
@@ -145,22 +145,21 @@ func adjustInt64(v *int64, defValue int64) {
 
 // Parse parses flag definitions from the argument list.
 func (c *Config) Parse(arguments []string) error {
-	flagArgs := make(map[string]*flag.Flag)
-	flag.Visit(func(flag *flag.Flag) {
-		flagArgs[flag.Name] = flag
-	})
-
-	var err error
-	if confFlag, ok := flagArgs["config"]; ok {
-		c.configFile = confFlag.Value.String()
-		if c.configFile != "" {
-			err = c.configFromFile(c.configFile)
-			if err != nil {
-				return errors.Trace(err)
-			}
+	// Parse first to get config file.
+	err := c.FlagSet.Parse(arguments)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
```

Comment:
I think we only do a trick to get the config file here, we should leave the check at the end.
I don't think we should care performance here.